shell/platform/darwin/macos/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -629,6 +629,26 @@ - (bool)testTrackpadGesturesAreSentToFramework {
   EXPECT_EQ(last_event.scroll_delta_x, -40 * viewController.flutterView.layer.contentsScale);
   EXPECT_EQ(last_event.scroll_delta_y, -80 * viewController.flutterView.layer.contentsScale);
 
+  // A discrete scroll event should use the PointerSignal system, and flip the
+  // direction when shift is pressed.
+  CGEventRef cgEventDiscreteShift =
+      CGEventCreateScrollWheelEvent(NULL, kCGScrollEventUnitPixel, 1, 0);
+  CGEventSetType(cgEventDiscreteShift, kCGEventScrollWheel);
+  CGEventSetFlags(cgEventDiscreteShift, kCGEventFlagMaskShift);
+  CGEventSetIntegerValueField(cgEventDiscreteShift, kCGScrollWheelEventIsContinuous, 0);
+  CGEventSetIntegerValueField(cgEventDiscreteShift, kCGScrollWheelEventDeltaAxis2,
+                              1);  // scroll_delta_x
+  CGEventSetIntegerValueField(cgEventDiscreteShift, kCGScrollWheelEventDeltaAxis1,
+                              2);  // scroll_delta_y
+
+  called = false;
+  [viewController scrollWheel:[NSEvent eventWithCGEvent:cgEventDiscreteShift]];
+  EXPECT_TRUE(called);
+  EXPECT_EQ(last_event.signal_kind, kFlutterPointerSignalKindScroll);
+  // pixelsPerLine is 40.0, direction is reversed and axes have been flipped.
+  EXPECT_EQ(last_event.scroll_delta_x, -80 * viewController.flutterView.layer.contentsScale);
+  EXPECT_EQ(last_event.scroll_delta_y, -40 * viewController.flutterView.layer.contentsScale);
```

Comment:
Looks like we're already being naughty in this file, but we should be using `EXPECT_FLOAT_EQ`/`EXPECT_DOUBLE_EQ` for floats and doubles.